hamster_gtk/tracking/screens.py

```diff
@@ -178,13 +188,105 @@ def __init__(self, controller, *args, **kwargs):
 
         # Fact entry field
         self.raw_fact_entry = Gtk.Entry()
+        self.raw_fact_entry.set_completion(self._get_completion())
         self.pack_start(self.raw_fact_entry, False, False, 0)
 
         # Buttons
         start_button = Gtk.Button(label=_('Start Tracking'))
         start_button.connect('clicked', self._on_start_tracking_button)
         self.pack_start(start_button, False, False, 0)
 
+    def _get_completion(self):
+        """
+        Return a completion instance to match 'activity@category' strings.
+
+        Returns:
+            Gtk.EntryCompletion: Completion instance.
+        """
+        def remove_duplicates(list):
+            """
+            Quick and dirty function to remove duplicate list items.
+
+            The first item encounterd is keept, the following thrown away.
+
+            We can not simply turn this into a set as we would loose the
+            ordering.  This clearly is highly inefficiennt and the only reason
+            we do this for now is the very limited size of the input list.
+
+            For one possible way to do this see: ``https://code.activestate.com/recipes/576694/``.
```

Comment:
There is an efficient OrderedSet implementation on PyPi: https://pypi.python.org/pypi/orderedset